src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -252,6 +477,215 @@ private TemporaryPoint createNewPoint(final Coordinate coordinate,
         return tags;
     }
 
+    /**
+     * Creates a mapping of all inner rings enclosed by the outer rings.
+     *
+     * @param outerToInnerMap
+     *            The mapping to update any time we find an outer ring that contains an inner ring
+     * @param outerRings
+     *            The list of outer {@link LinearRing}s to use
+     * @param innerRings
+     *            The list of inner {@link LinearRing}s to use
+     */
+    private void findEnclosedRings(final MultiMap<Integer, Integer> outerToInnerMap,
+            final List<LinearRing> outerRings, final List<LinearRing> innerRings)
+    {
+        for (int innerIndex = 0; innerIndex < innerRings.size(); innerIndex++)
+        {
+            final LinearRing inner = innerRings.get(innerIndex);
+            boolean foundEnclosingOuter = false;
+            for (int outerIndex = 0; outerIndex < outerRings.size(); outerIndex++)
+            {
+                final LinearRing outer = outerRings.get(outerIndex);
+                final com.vividsolutions.jts.geom.Polygon outerPolygon = new com.vividsolutions.jts.geom.Polygon(
+                        outer, null, JtsUtility.GEOMETRY_FACTORY);
+                if (outerPolygon.contains(inner))
+                {
+                    foundEnclosingOuter = true;
+                    outerToInnerMap.add(outerIndex, innerIndex);
+                }
+            }
+
+            if (!foundEnclosingOuter)
+            {
+                // Isolated inner ring, invalid multipolygon
+                logger.error("Found isolated inner member for Multipolygon Relation geometry: {}",
+                        inner);
+            }
+        }
+    }
+
+    /**
+     * @param closedOuterLines
+     *            The list of closed outer {@link Line}s to use
+     * @param closedInnerLines
+     *            The list of closed inner {@link Line}s to use
+     * @return a mapping of intersecting outer to inner relation members
+     */
+    private MultiMap<Integer, Integer> findIntersectingMembers(final List<Line> closedOuterLines,
+            final List<Line> closedInnerLines)
+    {
+        final MultiMap<Integer, Integer> outerToInnerIntersectionMap = new MultiMap<>();
+
+        if (!closedOuterLines.isEmpty() && !closedInnerLines.isEmpty())
+        {
+            for (int innerIndex = 0; innerIndex < closedInnerLines.size(); innerIndex++)
```

Comment:
Maybe hoist `closedInnerLines.size()` out as a temporary variable since it's used a few times in the loop and looks to be invariant at first glance.

Reply:
I only see it used once. Do you mean pull out the sizes of the outer and inner lines? If I remember correctly, ArrayList does this under the covers for us already.